src/elements/fields/DateSelectorField/index.tsx

```diff
@@ -286,3 +288,37 @@ function DateSelectorField({
 }
 
 export default memo(DateSelectorField);
+
+const CustomMaskedInput = React.forwardRef((props: any, ref) => {
+  const { onChange, dateMask, ...rest } = props;
+  const maskOptions = useMemo(
+    () => ({
+      mask: dateMask,
+      blocks: {
+        dd: { mask: IMask.MaskedRange, from: 1, to: 31, maxLength: 2 },
+        MM: { mask: IMask.MaskedRange, from: 1, to: 12, maxLength: 2 },
+        yyyy: { mask: IMask.MaskedRange, from: 1900, to: 2099 },
```

Comment:
do we want to restrict it to this range? what about 1 to 9999?

Reply:
Yeah good call. I made the range 1-9999.